pkg/components/dashdiffs/formatter_json.go

```diff
@@ -161,23 +161,35 @@ func (f *JSONFormatter) Format(diff diff.Diff) (result string, err error) {
 func (f *JSONFormatter) formatObject(left map[string]interface{}, df diff.Diff) {
 	f.addLineWith(ChangeNil, "{")
 	f.push("ROOT", len(left), false)
-	f.processObject(left, df.Deltas())
+	if err := f.processObject(left, df.Deltas()); err != nil {
+		f.pop()
```

Comment:
Return error, but not sure about the `f.Pop()` since returning this error should probably fail, but unsure.

Reply:
@marefr Fixed to return error, but at least in general I should think we should clean up state (i.e., what got pushed) before returning the error?

I'm very used to programming with exception safety in mind, i.e. that one should strive to keep a system in a consistent state even if an exception is handled and execution continues instead of crashing.